sdk/core/azure-core/src/main/java/com/azure/core/util/ExpandableStringEnum.java

```diff
@@ -56,10 +62,13 @@ protected static <T extends ExpandableStringEnum<T>> T fromString(String name, C
             return value;
         } else {
             try {
-                value = clazz.newInstance();
+                MethodHandles.Lookup lookup = ReflectionUtils.getLookupToUse(clazz);
+                value = (T) lookup.findConstructor(clazz, methodType(void.class)).invoke();
                 return value.nameAndAddValue(name, value, clazz);
-            } catch (IllegalAccessException | InstantiationException ex) {
+            } catch (NoSuchMethodException | IllegalAccessException e) {
                 return null;
+            } catch (Throwable e) {
+                throw new RuntimeException(e);
```

Comment:
nit: We should inspect the type of Throwable, if this is an Error we shouldn't wrapped it and just re-throw it. Also if the Throwable is already a RuntimeException there is no need to wrap it either

Reply:
`Throwable` comes from `MethodHandle.invoke` definition and it's the other way around -  `RuntimeException` is `Throwable` so I have to wrap it or log and return null. Previously we returned null if something didn't go well, so I switched back to this behavior and added some logs.